gnocchi/storage/_carbonara.py

```diff
@@ -117,9 +118,15 @@ def _store_metric_measures(metric, timestamp_key, aggregation,
                                granularity, data, offset=None, version=3):
         raise NotImplementedError
 
-    @staticmethod
-    def _list_split_keys_for_metric(metric, aggregation, granularity,
+    def _list_split_keys_for_metric(self, metric, aggregation, granularity,
                                     version=3):
+        return set(map(
+            functools.partial(carbonara.SplitKey, sampling=granularity),
+            utils.to_timestamps(self._list_split_keys(
```

Comment:
What I can do is to change the test in this patch so rather than having:

`+            carbonara.SplitKey(numpy.datetime64('2016-01-10T00:00:00'), 60),`

we'll have
`+            carbonara.SplitKey(1452384000.0, 60),`

and then the workaround is not needed.

But then i'll have to rechange to

`+            carbonara.SplitKey(numpy.datetime64('2016-01-10T00:00:00'), 60),`

in a later patch.

Is this worth my time? :)